src/NonLinearProgram/nlp_utilities.jl

```diff
@@ -0,0 +1,501 @@
+# Copyright (c) 2025: Andrew Rosemberg and contributors
+#
+# Use of this source code is governed by an MIT-style license that can be found
+# in the LICENSE.md file or at https://opensource.org/licenses/MIT.
```

Comment:
the header should mention the tutorial you got code from

Reply:
done